src/async_monitored_item.rs

```diff
@@ -254,10 +259,12 @@ fn delete_monitored_items(client: &ua::Client, request: &ua::DeleteMonitoredItem
         // does not take ownership.
         let request = unsafe { ua::DeleteMonitoredItemsRequest::to_raw_copy(request) };
 
+        let Ok(mut client) = client.lock() else {
+            panic!("mutex should not have been poisoned");
+        };
         unsafe {
             UA_Client_MonitoredItems_delete_async(
-                // SAFETY: Cast to `mut` pointer, function is marked `UA_THREADSAFE`.
```

Comment:
Shouldn't we keep those comments and instead mention at the workaround `Mutex` why they (still) exist? Their message still applies, even though we lock the outer mutex as a band-aid.